blueprints/blueprint/index.js

```diff
@@ -1,13 +1,45 @@
 module.exports = {
   description() {
-    return 'Generates a blueprint template for creating new blueprints';
+    return 'Generates a blueprint with desired hooks';
```

Comment:
Sure. That's mostly yargs stuff, which I could link to in a comment, but I agree it serves as a useful example.

Reply:
Upon investigation, it is also parsing command line args like so:
`blueprint_bp__blueprint__afterHook=false bp config --bp.blueprint.command=true` produces
`{"bp":{"blueprint":{"afterHook": false, "command": true}}`

We can turn off argv processing for rc and let yargs handle it.  Can yargs allow arbitrary options?  would I have to define '--bp.partials.includeDir=some/path` and `--bp.partials.onlyPartials=true` explicity as options?

Reply:
yargs can indeed allow arbitrary values. Or you can tell it not to. That could be decided/enforced on a command-by-command (blueprint-by-blueprint) basis. 

I definitely agree that blueprint options and the rc files(s) need to be integrated. I'm open to discussion how that is best achieved. At the moment I'm leaning towards cleaner/simpler options in the blueprints themselves. If we offer the "full power" of yargs to blueprint authors it's unpleasant to mandate that they define them in --nested.dot.object format to map directly to the structure of the rc file(s). And makes them brittle if we change that format.

I've been thinking more along the lines of extracting the blueprint specific config from the resolved rc config and passing that to yargs (and maybe the options builder so it can relay them to the user). Possibly by setting a config property on the blueprint at load time. This would allow rc to handle environment variables to customise the nested structure but individual commands to specify simple options.

I'm more than happy to go with the consensus. Perhaps we could arrange a real-time discussion to get to the bottom of it?